tools/cdk-release/lib/lifecycles/changelog.ts

```diff
@@ -0,0 +1,156 @@
+      repoUrl: `${host}/${owner}/${repository}`,
+      linkCompare: true,
+      previousTag: `v${currentVersion}`,
+      currentTag: `v${newVersion}`,
+      // when isPatch is 'true', the default template used for the header renders an H3 instead of an H2
+      // (see: https://github.com/conventional-changelog/conventional-changelog/blob/f1f50f56626099e92efe31d2f8c5477abd90f1b7/packages/conventional-changelog-conventionalcommits/templates/header.hbs#L1-L5)
+      isPatch: false,
+    };
+    // invoke the conventionalChangelogWriter package that will perform the actual Changelog rendering
+    const changelogStream = commitsStream
+      .pipe(conventionalChangelogWriter(context,
+        {
+          // CDK uses the settings from 'conventional-changelog-conventionalcommits'
+          // (by way of 'standard-version'),
+          // which are different than the 'conventionalChangelogWriter' defaults
+          ...presetConfig.writerOpts,
+          finalizeContext: (ctx: { noteGroups?: { title: string }[], date?: string }) => {
+            // the heading of the "BREAKING CHANGES" section is governed by this Handlebars template:
+            // https://github.com/conventional-changelog/conventional-changelog/blob/f1f50f56626099e92efe31d2f8c5477abd90f1b7/packages/conventional-changelog-conventionalcommits/templates/template.hbs#L3-L12
+            // to change the heading from 'BREAKING CHANGES' to 'BREAKING CHANGES TO EXPERIMENTAL FEATURES',
+            // we have to change the title of the 'BREAKING CHANGES' noteGroup
+            ctx.noteGroups?.forEach(noteGroup => {
+              if (noteGroup.title === 'BREAKING CHANGES') {
+                noteGroup.title = 'BREAKING CHANGES TO EXPERIMENTAL FEATURES';
+              }
+            });
+            // in unit tests, we don't want to have the date in the Changelog
+            if (args.includeDateInChangelog === false) {
+              ctx.date = undefined;
+            }
+            return ctx;
+          },
+        }));
+
+    changelogStream.on('error', function (err: any) {
+      reject(err);
+    });
+    let content = '';
+    changelogStream.on('data', function (buffer: any) {
+      content += buffer.toString();
+    });
+    changelogStream.on('end', function () {
+      checkpoint(args, 'outputting changes to %s', [args.infile]);
+      if (args.dryRun) {
+        debug(args, `\n---\n${content.trim()}\n---\n`);
+      } else {
+        writeFile(args, args.infile!, args.changeLogHeader + '\n' + (content + oldContent).replace(/\n+$/, '\n'));
+      }
+      return resolve(content);
+    });
+  });
+}
+
+function createChangelogIfMissing(args: ReleaseOptions) {
+  if (!fs.existsSync(args.infile!)) {
+    checkpoint(args, 'created %s', [args.infile]);
+    // args.outputUnreleased = true
+    writeFile(args, args.infile!, '\n');
+  }
+}
+
+function stripExperimentalCommitsIfNeeded(args: ReleaseOptions, commits: ConventionalCommit[]): ConventionalCommit[] {
```

Comment:
Ooooohhh I see whatcha doooooin'.

I think I would like the fact that this happens exposed at the top level, in `index.ts`. It's a bit hidden that this happens DURING changelog generation, but it's pretty fundamental to this tool that we have the ability to do this at all. This kind of logic is also the logic we might change depending on our future directions with releasing and the experimental packages, so it makes sense to call them out top-level.

The top level should look something like this, imo:

```ts
const commits = readCommits();
const filteredCommits = filterCommits(comments);
writeChangeLog(filteredCommits);
```

Or something

Reply:
Sure. Changed.